### PR TITLE
[Product] Refactor product variant generator

### PR DESCRIPTION
**Pull request by pamil**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | yes
| Related tickets | -
| License         | MIT

- [x] Removed `$isRecursiveStep` argument from public method `build()`
- [x] Removed `SetBuilderInterface` as it is really hard to enforce a contract there (what it takes and what it returns)
- [x] Put `CartesianSetBuilder` directly into the default `ProductVariantGenerator`

### CODE REVIEW

File: src/Sylius/Component/Product/Generator/CartesianSetBuilder.php [hunk 1]
```diff
@@ -9,22 +9,35 @@
...
+        return $this->doBuild($setTuples, false);
+    }
+
+    /**
+     * @param array $setTuples
+     * @param bool $isRecursiveStep
+     *
+     * @return array
+     */
+    private function doBuild(array $setTuples, $isRecursiveStep)
```
Comment (lchrusciel) [inline src/Sylius/Component/Product/Generator/CartesianSetBuilder.php]: Instead of `doBuild` we should name it `buildRecursive`. It will be more meaningful.

Comment (Zales0123) [inline src/Sylius/Component/Product/Generator/CartesianSetBuilder.php]: DO
DO BUILD
DO BUILD MICH
do-build

File: src/Sylius/Component/Product/Generator/CartesianSetBuilder.php [hunk 2]
```diff
@@ -9,22 +9,35 @@
...
-namespace Sylius\Component\Product\SetBuilder;
+namespace Sylius\Component\Product\Generator;
 
 /**
  * Builds the Cartesian product set from one or more given sets.
  *
  * @author Paweł Jędrzejewski <pawel@sylius.org>
  */
-final class CartesianSetBuilder implements SetBuilderInterface
+final class CartesianSetBuilder
```
Comment (lchrusciel) [inline src/Sylius/Component/Product/Generator/CartesianSetBuilder.php]: Why did you remove interface? I guess cartesian set can be generated different way, based on a chosen algorithm

Comment (lchrusciel) [inline src/Sylius/Component/Product/Generator/CartesianSetBuilder.php]: Ok, I get it. Im not sure about that, but ok, description of PR makes sense.

File: src/Sylius/Component/Product/Generator/CartesianSetBuilder.php (hunk 1 above)
Comment (pamil) [inline src/Sylius/Component/Product/Generator/CartesianSetBuilder.php]: The first call made in `build()` method is not recursive :)

**Comment by pjedrzejewski** [comment]
Thank you Kamil!
